client/customer-effort-score-tracks/data/reducer.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * Internal dependencies
+ */
+import TYPES from './action-types';
+
+const DEFAULT_STATE = {
+	queue: [],
+};
+
+const reducer = ( state = DEFAULT_STATE, action ) => {
+	switch ( action.type ) {
+		case TYPES.SET_CES_SURVEY_TRACKS_QUEUE:
+			return {
+				...state,
+				queue: action.queue,
+			};
+		case TYPES.ADD_CES_SURVEY_TRACK:
+			// Prevent duplicate
+			const registeredTracks = new Set(
+				state.queue.map( ( track ) => track.action )
+			);
+			if ( registeredTracks.has( action.action ) ) {
+				return state;
+			}
```

Comment:
Why map to a `Set` first vs just using the queue directly to find if a duplicate action is in the queue?

Reply:
That's a good catch 👍 I've updated it in https://github.com/woocommerce/woocommerce-admin/pull/5649/commits/9735e64b91d7762a342f136da051d5bf8acdcf6b